workcal/angular/src/app/calendar-page/calendar-page.component.ts

```diff
@@ -0,0 +1,482 @@
+    //this.router.navigate([this.router.url]);
+  }
+
+
+
+  fetchEvents(): void {
+    this.eventApiService.getAllEvents()
+      .pipe(
+        catchError(error => {
+          console.error('Error fetching events:', error);
+          return of([]);
+        })
+      )
+      .subscribe(data => {
+        this.events = data;
+        this.schedulerEvents = data.map(event => ({
+          id: event.id,
+          startDate: event.startTime,
+          endDate: event.endTime,
+          text: event.name,
+          location: event.location,
+          labels: event.labels,
+          users: event.users.map(user => ({ id: user.id, userName: user.userName }))
+        }));
+      });
+
+  }
+
+
+
+  fetchUsers(): void {
+    this.userApiService.getAllUsers()
+      .pipe(
+        catchError(error => {
+          console.error('Error fetching users:', error);
+          return of([]);
+        })
+      )
+      .subscribe((data: UserResponse | any[]) => { // Explicitly type data
+        if (Array.isArray(data)) {
+          console.error('Received an array, expected an object with an items key:', data);
+          return;
+        }
+
+        if (data && data.items) {
+          this.allusers = data.items.map(user => ({
+            id: user.id,
+            userName: user.userName,
+            name: user.name,
+            email: user.email
+          }));
+        } else {
+          console.error('Items key not found in response:', data);
+        }
+      });
+
+  }
+
+
+  createEvent(newEvent: EventDto): void {
+    this.eventApiService.createEvent(newEvent).subscribe(
+      (response) => {
+        this.fetchEvents();
+      },
+      (error) => {
+        console.error('Error creating event:', error);
+      }
+    );
+    console.log(this.allusers);
+    this.refreshPage();
+
+  }
+
+
+  updateEvent(eventToUpdate: EventDto): void {
+    this.eventApiService.updateEvent(eventToUpdate.id, eventToUpdate).subscribe( () => {
+      this.fetchEvents();
+    },
+    (error) => {
+      console.error('Error updating event:', error);
+    }
+
+  );
+  this.fetchEvents();
+  this.refreshPage();
+
+  }
+
+  deleteEvent(eventId: string): void {
+    const index = this.events.findIndex(event => event.id === eventId);
+    if (index !== -1) {
+      this.eventApiService.deleteEvent(eventId).subscribe(() => {
+        this.events.splice(index, 1);
+        this.fetchEvents();
+      });
+    }
+    this.refreshPage();
+
+  }
+
+  onEventAdding(event): void {
+    const appointmentData = event.appointmentData;
+
+    const selectedLabels = this.defaultLabels.filter(label =>
+      appointmentData.labels?.includes(label.name)
+    );
+
+      const selectedUsers = appointmentData.users.filter(user =>
+        appointmentData.users?.includes(user.name)
+      );
+
+      const selectedUserIDs = appointmentData.users;
+
+
+
+    const newEvent: EventDto = {
+      name: appointmentData.text,
+      startTime: new Date(appointmentData.startDate),
```

Comment:
the dates we can generate in backend we don't need this here.
In backend side it will be generated other way which more safer regarding the timezones.
Bad practice to do this here.